paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentOptionsActivityTest.kt

```diff
@@ -285,80 +281,64 @@ internal class PaymentOptionsActivityTest {
 
         val viewModel = createViewModel(args)
 
-        val transitionTargets = mutableListOf<TransitionTarget>()
-        viewModel.transition.observeEventsForever { transitionTargets.add(it) }
-
-        activityScenario(viewModel).launch(createIntent(args)).use {
-            it.onActivity {
-                idleLooper()
-            }
+        viewModel.currentScreen.test {
+            assertThat(awaitItem()).isNull()
+            activityScenario(viewModel).launch(createIntent(args)).onActivity { idleLooper() }
+            assertThat(awaitItem()).isEqualTo(TransitionTarget.SelectSavedPaymentMethods)
         }
-
-        assertThat(transitionTargets).containsExactly(TransitionTarget.SelectSavedPaymentMethods)
     }
 
     @Test
-    fun `Verify if customer has payment methods, display the saved payment methods screen`() {
+    fun `Verify if customer has payment methods, display the saved payment methods screen`() = runTest {
         val args = PAYMENT_OPTIONS_CONTRACT_ARGS.updateState(
             isGooglePayReady = false,
             paymentMethods = listOf(PaymentMethodFixtures.CARD_PAYMENT_METHOD)
         )
 
         val viewModel = createViewModel(args)
 
-        val transitionTargets = mutableListOf<TransitionTarget>()
-        viewModel.transition.observeEventsForever { transitionTargets.add(it) }
-
-        activityScenario(viewModel).launch(createIntent(args)).use {
-            idleLooper()
+        viewModel.currentScreen.test {
+            assertThat(awaitItem()).isNull()
+            activityScenario(viewModel).launch(createIntent(args)).onActivity { idleLooper() }
```

Comment:
Not a big deal either way, but I'm mostly curious. Is this `idleLooper` required? Could you use `advanceUntilIdle()` from coroutine land instead?

Not really sure there's a benefit either way, mostly just curious.

Reply:
Seems like removing it doesn’t break things 🤷‍♂️